triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerLogContext.scala

```diff
@@ -51,24 +51,44 @@ final class TriggerLogContext private (
 
   import ToLoggingContext._
 
+  private[this] var callback: (String, TriggerLogContext) => Unit = (_, _) => ()
+
+  private[trigger] def setCallback(callback: (String, TriggerLogContext) => Unit): Unit = {
+    this.callback = callback
+  }
+
   def enrichTriggerContext[A](
       additionalEntries: (String, LoggingValue)*
   )(f: TriggerLogContext => A): A = {
-    f(new TriggerLogContext(loggingContext, entries ++ additionalEntries, span))
+    val context = new TriggerLogContext(loggingContext, entries ++ additionalEntries, span)
+
+    context.setCallback(callback)
+
+    f(context)
   }
 
   def nextSpan[A](
       name: String,
       additionalEntries: (String, LoggingValue)*
   )(f: TriggerLogContext => A): A = {
-    f(new TriggerLogContext(loggingContext, entries ++ additionalEntries, span.nextSpan(name)))
+    val context =
+      new TriggerLogContext(loggingContext, entries ++ additionalEntries, span.nextSpan(name))
+
+    context.setCallback(callback)
+
+    f(context)
   }
 
   def childSpan[A](
       name: String,
       additionalEntries: (String, LoggingValue)*
   )(f: TriggerLogContext => A): A = {
-    f(new TriggerLogContext(loggingContext, entries ++ additionalEntries, span.childSpan(name)))
+    val context =
+      new TriggerLogContext(loggingContext, entries ++ additionalEntries, span.childSpan(name))
+
+    context.setCallback(callback)
```

Comment:
We need to ensure all newly created logging contexts have the current callback defined upon them